src/test/java/com/github/fridujo/rabbitmq/mock/ChannelTest.java

```diff
@@ -633,4 +632,107 @@ void commit_or_rollback_can_be_called_multiple_times_after_a_single_select() thr
+                assertThat(channel.messageCount(queue)).isEqualTo(1);
+
+                final GetResponse basicGet = channel.basicGet(queue, true);
+                final String replyTo = basicGet.getProps().getReplyTo();
+                assertThat(replyTo).startsWith("amq.gen-");
+
+                channel.basicPublish("", replyTo, null, "pong".getBytes());
+
+                CountDownLatch latch = new CountDownLatch(1);
+                AtomicBoolean cancelled = new AtomicBoolean();
+                AtomicReference<String> reply = new AtomicReference<String>();
+                String consumerTag = channel.basicConsume("amq.rabbitmq.reply-to", true, new DefaultConsumer(channel) {
+                    @Override
+                    public void handleDelivery(String consumerTag,
+                                               Envelope envelope,
+                                               AMQP.BasicProperties properties,
+                                               byte[] body) {
+                        assertThat(reply.compareAndSet(null, new String(body))).isTrue();
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void handleCancelOk(String consumerTag) {
+                        cancelled.set(true);
+                    }
+                });
+
+                latch.await(1, TimeUnit.SECONDS);
+                channel.basicCancel(consumerTag);
+
+                assertThat(cancelled).isTrue();
+                assertThat(reply.get()).isEqualTo("pong");
+            }
+        }
+    }
+
+    @Test
+    void directReplyTo_basicConsume_noAutoAck() {
+        assertThrows(IllegalStateException.class, () -> {
+            try (Connection conn = new MockConnectionFactory().newConnection()) {
+                try (Channel channel = conn.createChannel()) {
+                    channel.basicConsume("amq.rabbitmq.reply-to", false, new DefaultConsumer(channel) {
+                        @Override
+                        public void handleDelivery(String consumerTag,
+                                                   Envelope envelope,
+                                                   AMQP.BasicProperties properties,
+                                                   byte[] body) {
+
+                            fail();
+                        }
+
+                        @Override
+                        public void handleCancelOk(String consumerTag) {
+                            fail();
+                        }
+                    });
+                }
+            }
+        });
+    }
+
+    @Test
+    void directReplyTo_basicGet_noAutoAck() {
+        assertThrows(IllegalStateException.class, () -> {
+            try (Connection conn = new MockConnectionFactory().newConnection()) {
+                try (Channel channel = conn.createChannel()) {
+                    final GetResponse reply = channel.basicGet("amq.rabbitmq.reply-to", false);;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/app/ledoyen/rabbitmq-mock/pullRequest?prid=4732226)